src/prefs.js

```diff
@@ -1,34 +1,43 @@
-import Gio from 'gi://Gio';
-import Adw from 'gi://Adw';
-
-import {ExtensionPreferences, gettext as _} from 'resource:///org/gnome/Shell/Extensions/js/extensions/prefs.js';
+import Gio from "gi://Gio";
+import Adw from "gi://Adw";
 
+import {ExtensionPreferences, gettext as _} from "resource:///org/gnome/Shell/Extensions/js/extensions/prefs.js";
 
 export default class ContainersPreferences extends ExtensionPreferences {
     fillPreferencesWindow(window) {
-        // Create a preferences page, with a single group
+        window._settings = this.getSettings();
+
         const page = new Adw.PreferencesPage({
-            title: _('General'),
-            icon_name: 'dialog-information-symbolic',
+            title: _("General"),
+            icon_name: "dialog-information-symbolic",
         });
         window.add(page);
 
-        const group = new Adw.PreferencesGroup({
-            title: _('Appearance'),
-            description: _('Configure the appearance of the extension'),
+        const appearanceGroup = new Adw.PreferencesGroup({
+            title: _("Appearance"),
+            description: _("Configure the appearance of the extension"),
         });
-        page.add(group);
+        page.add(appearanceGroup);
 
-        // Create a new preferences row
-        const row = new Adw.SwitchRow({
-            title: _('Extra Info'),
-            subtitle: _('Whether to show extra info of a container in name and the opened menu'),
+        const extraInfoRow = new Adw.SwitchRow({
+            title: _("Extra Info"),
+            subtitle: _("Whether to show extra info of a container in name and the opened menu"),
         });
-        group.add(row);
+        appearanceGroup.add(extraInfoRow);
+        window._settings.bind("extra-info", extraInfoRow, "active", Gio.SettingsBindFlags.DEFAULT);
 
-        // Create a settings object and bind the row to the `extra-info` key
-        window._settings = this.getSettings();
-        window._settings.bind('extra-info', row, 'active',
-            Gio.SettingsBindFlags.DEFAULT);
+        const behaviourGroup = new Adw.PreferencesGroup({
+            title: _("Behaviour"),
+            description: _("Configure the behaviour of the extension"),
+        });
+        page.add(behaviourGroup);
+
+        const terminalRow = new Adw.EntryRow({
+            title: _("Terminal program with arguments"),
+            show_apply_button: true, // Allows user to apply the input
+        });
+        behaviourGroup.add(terminalRow);
+        window._settings.bind("terminal", terminalRow, "text", Gio.SettingsBindFlags.DEFAULT);
```

Comment:
Just a silly question

Does gnome extension use mvvm to set the config, and we only need to set a new schema of the config then it just automatically has storage for saving it? 

I seen on https://gjs.guide/extensions/development/preferences.html#gsettings 

But well I never code it, just making sure. 

Thank you

Reply:
I don't how the internal backend system for gsettengs works, I just use the API. It is used for a lot other gnome related configurations, not only for extensions though.